cql3/restrictions/statement_restrictions.hh

```diff
@@ -414,7 +414,11 @@ public:
     }
 
     bool ck_restrictions_need_filtering() const {
-        return _partition_key_restrictions->has_unrestricted_components(*_schema) || _clustering_columns_restrictions->needs_filtering(*_schema);
+        return _partition_key_restrictions->has_unrestricted_components(*_schema) 
+            // In TOKEN restrictions on tables with secondary indexes we only restrict idx_token column in view,
+            // leaving all partition key columns unrestricted.
+            || (_uses_secondary_indexing && has_token(_partition_key_restrictions->expression)) 
```

Comment:
Technically, you also need `&& !_clustering_columns_restrictions->empty()` here; otherwise we may have an absurd result of no CK restrictions but ck_restrictions_need_filtering() returns true.  The current caller happens not to care, but it's better to be future-proof.

Reply:
Maybe it would make more sense to just rename this method to `should_skip_ck_restrictions()`?

It is used only in those 4 places:

![Zrzut ekranu z 2020-11-16 12-14-10](https://user-images.githubusercontent.com/10003183/99246320-663aa900-2805-11eb-980c-3f53dd934fd4.png)

For me it makes more sense the way it works now, rather than making it more "general" by adding `&& !_clustering_columns_restrictions->empty()`.

Reply:
@avelanarius please use text quoting, not gifs, they are searchable and don't trigger allergies in old people.

Reply:
> Maybe it would make more sense to just rename this method to `should_skip_ck_restrictions()` (and negate it)?

I'd be OK with that.  Especially if the method's declaration included a good description.

Reply:
Having fixed it according to the `should_skip_*` idea, I ultimately decided to do the `&& !_clustering_columns_restrictions->empty()` fix, as it proved to be more elegant. (changed to it in amended commit).